tests/queries/0_stateless/02870_per_column_settings.sql

```diff
@@ -0,0 +1,78 @@
+-- Tags: no-random-merge-tree-settings
+
+-- Tests column-level settings for MergeTree* tables
+
+DROP TABLE IF EXISTS tab;
+
+CREATE TABLE tab
+(
+    id UInt64,
+    long_string String SETTINGS (min_compress_block_size = 163840, max_compress_block_size = 163840),
+    v1 String,
+    v2 UInt64,
+    v3 Float32,
+    v4 Float64
+)
+ENGINE = ReplicatedMergeTree('/clickhouse/tables/{database}/tab/2870', 'r1')
+ORDER BY id
+SETTINGS min_bytes_for_wide_part = 1;
+
+SHOW CREATE tab;
+
+INSERT INTO TABLE tab SELECT number, randomPrintableASCII(1000), randomPrintableASCII(10), rand(number), rand(number+1), rand(number+2) FROM numbers(1000);
+SELECT count() FROM tab;
+
+ALTER TABLE tab MODIFY COLUMN long_string MODIFY SETTING min_compress_block_size = 8192;
+SHOW CREATE tab;
+
+ALTER TABLE tab MODIFY COLUMN long_string RESET SETTING min_compress_block_size;
+SHOW CREATE tab;
+
+ALTER TABLE tab MODIFY COLUMN long_string REMOVE SETTINGS;
+SHOW CREATE tab;
+
+ALTER TABLE tab MODIFY COLUMN long_string String SETTINGS (min_compress_block_size = 163840, max_compress_block_size = 163840);
+SHOW CREATE tab;
```

Comment:
Up to you but perhaps the tests should also check that column-level settings really have an effect on compression (except for a new CREATE statement returned by SHOW CREATE).

EDIT: On the other hand, column sizes differ slightly between platforms, so if this makes this test not runnable on ARM then we can level as is.